editoast/src/views/timetable/stdcm.rs

```diff
@@ -191,6 +185,31 @@ async fn stdcm(
     let earliest_departure_time = stdcm_request.get_earliest_departure_time(simulation_run_time);
     let latest_simulation_end = stdcm_request.get_latest_simulation_end(simulation_run_time);
 
+    let timetable = Timetable::retrieve_or_fail(&mut conn, timetable_id, || {
+        StdcmError::TimetableNotFound { timetable_id }
+    })
+    .await?;
+    // Filter train
+    let mut train_schedules =
```

Comment:
Since the timetable is already fetched, wdyt about an associated method?

```suggestion
    let mut train_schedules = timetable.schedules_in_time_window(earliest_departure_time, latest_simulation_end).await?;
```